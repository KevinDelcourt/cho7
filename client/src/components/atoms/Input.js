import styled from "styled-components"

const Input = styled.input`
    border-radius: 5px;
    border: 0;
    padding: 5px;
`

export default Input
