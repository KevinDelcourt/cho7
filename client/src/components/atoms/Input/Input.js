import styled from "styled-components"

const Input = styled.input`
    border-radius: ${props => props.borderRadius || "5px"};
    padding: 5px;
    margin-bottom: 7px;
`

export default Input
