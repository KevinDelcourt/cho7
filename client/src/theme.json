{
    "colors":{
        "connectionButton":"#916D43",
        "menuButton":"rgba(145, 109, 67, 0.35)",
        "submitButton":"#C4C4C4",
        "mainContainer": "rgba(145, 109, 67, 0.35)"
    },
    "border":{
        "mainContainerCorners":"20px",
        "mainContainer":"none"
    }

}