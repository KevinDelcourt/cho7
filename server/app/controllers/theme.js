const { queryCallback } = require("../../modules/queries")

const themeController = connection => {
    return {
        getTheme: (req, res) => {
            connection.query("SELECT * FROM theme", (err, rows) =>
                queryCallback(err, res, () => {
                    let theme = {}
                    rows.map(r => {
                        theme[r.style] = r.value
                    })
                    return res.send(theme)
                })
            )
        },

        updateTheme: (req, res) => {
            let query = {
                mysql: "INSERT INTO theme (style,value) VALUES ",
                data: []
            }

            const addValues = (key, value) => {
                query.mysql += "(?,?),"
                query.data.push(key)
                query.data.push(value)
            }

            for (let key in req.body)
                if (req.body[key] != "undefined") addValues(key, req.body[key])

            for (let key in req.files)
                addValues(key.slice(0, -4), req.files[key][0].originalname)

            query.mysql = query.mysql.slice(0, -1)
            connection.query("DELETE FROM theme", (err, rows) => {
                queryCallback(err, res, () => {
                    connection.query(query.mysql, query.data, (err, rows) => {
                        queryCallback(err, res, () => res.send(true))
                    })
                })
            })
        }
    }
}

module.exports = themeController
