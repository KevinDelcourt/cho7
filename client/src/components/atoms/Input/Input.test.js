import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Input from "./Input"
import theme from "../../../theme.json"

test("Input border-radius renders correctly", () => {
    /*const tree = renderer.create(<Input borderRadius="10px" />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("border-radius", "10px")*/
})
