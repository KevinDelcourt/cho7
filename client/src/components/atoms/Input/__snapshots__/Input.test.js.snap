// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input border-radius renders correctly 1`] = `
.c0 {
  border-radius: 10px;
  padding: 5px;
  margin-top: 5px;
  margin-bottom: 7px;
}

<input
  className="c0"
/>
`;
